chain/client/src/view_client.rs

```diff
@@ -374,23 +526,24 @@ impl ViewClientActor {
             }
         }
 
-        let head = self.chain.head()?;
-        let target_shard_id = self
-            .runtime_adapter
+        let chain = self.chain(temp);
+        let runtime_adapter = self.runtime_adapter(temp);
+        let head = chain.head()?;
+        let target_shard_id = runtime_adapter
```

Comment:
And then cold runtime adapter can properly find shard_id and we don't need anything special in all of these calls that read EpochInfo inside.